examples/onoff_light_bt/src/main.rs

```diff
@@ -0,0 +1,387 @@
+use rs_matter::data_model::system_model::descriptor;
+use rs_matter::error::Error;
+use rs_matter::fabric::FabricMgr;
+use rs_matter::mdns::MdnsService;
+use rs_matter::pairing::DiscoveryCapabilities;
+use rs_matter::persist::Psm;
+use rs_matter::respond::DefaultResponder;
+use rs_matter::secure_channel::spake2p::VerifierData;
+use rs_matter::transport::core::MATTER_SOCKET_BIND_ADDR;
+use rs_matter::transport::network::btp::{Btp, BtpContext};
+use rs_matter::utils::buf::PooledBuffers;
+use rs_matter::utils::notification::Notification;
+use rs_matter::utils::select::Coalesce;
+use rs_matter::utils::std_mutex::StdRawMutex;
+use rs_matter::MATTER_PORT;
+
+mod comm;
+// TODO: Now that we have two examples, move common stuff to a `common` filder
+// The `dev_att` module would be a prime candidate for this.
+mod dev_att;
+
+static BTP_CONTEXT: BtpContext<StdRawMutex> = BtpContext::<StdRawMutex>::new();
+
+fn main() -> Result<(), Error> {
+    let thread = std::thread::Builder::new()
+        // Increase the stack size until the example can work without stack blowups.
+        // Note that the used stack size increases exponentially by lowering the level of compiler optimizations,
+        // as lower optimization settings prevent the Rust compiler from inlining constructor functions
+        // which often results in (unnecessary) memory moves and increased stack utilization:
+        // e.g., an opt-level of "0" will require a several times' larger stack.
+        //
+        // Optimizing/lowering `rs-matter` memory consumption is an ongoing topic.
+        .stack_size(200 * 1024)
+        .spawn(run)
+        .unwrap();
+
+    thread.join().unwrap()
+}
+
+fn run() -> Result<(), Error> {
+    env_logger::init_from_env(
+        env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+    );
+
+    info!(
+        "Matter memory: Matter={}B, IM Buffers={}B",
+        core::mem::size_of::<Matter>(),
+        core::mem::size_of::<PooledBuffers<10, NoopRawMutex, IMBuffer>>()
+    );
+
+    let dev_det = BasicInfoConfig {
+        vid: 0xFFF1,
+        pid: 0x8000,
+        hw_ver: 2,
+        sw_ver: 1,
+        sw_ver_str: "1",
+        serial_no: "aabbccdd",
+        device_name: "OnOff Light",
+        product_name: "Light123",
+        vendor_name: "Vendor PQR",
+    };
+
+    let dev_att = dev_att::HardCodedDevAtt::new();
+
+    let matter = Matter::new(
+        &dev_det,
+        &dev_att,
+        // NOTE:
+        // For `no_std` environments, provide your own epoch and rand functions here
+        MdnsService::Builtin,
+        rs_matter::utils::epoch::sys_epoch,
+        rs_matter::utils::rand::sys_rand,
+        MATTER_PORT,
+    );
+
+    let dev_comm = CommissioningData {
+        // TODO: Hard-coded for now
+        verifier: VerifierData::new_with_pw(123456, *matter.borrow()),
+        discriminator: 250,
+    };
+
+    //let discovery_caps = DiscoveryCapabilities::new(true, false, false);
```

Comment:
remove commented out code.
Booleans are usually hard to read in arguments (at least here ... i think it is easier when using an IDE with rust_analyzer). Could we restyle things somehow to make this easier to read (e.g. some builder pattern or flags with names)?

Reply:
Commented out code removed. As for restyling `DiscoveryCapabilities` - yes - changing it to `bitflags!` would make a lot of sense I think, because - if you look at its `as_bits` method - it is bitflags after all.

For now I've only left a comment about that on top of the type, as that would be beyond the scope of the BTP protocol we are introducing here.